parquet-hadoop/src/test/java/org/apache/parquet/statistics/RandomValueGenerators.java

```diff
@@ -0,0 +1,395 @@
+      BigInteger value = generateValue(minimum, maximum);
+      if (value != null) {
+        return value.toByteArray();
+      } else {
+        return null;
+      }
+    }
+  }
+
+
+  public static class RandomFloatGenerator extends RandomValueGenerator<Float> {
+    public RandomFloatGenerator(long seed, int total, float minimum, float maximum) {
+      super(seed, PrimitiveType.PrimitiveTypeName.FLOAT, total, minimum, maximum);
+    }
+
+    @Override
+    protected Float generateValue(Float minimum, Float maxiumum) {
+      float range = (maxiumum - minimum);
+
+      if (!shouldGenerateNull()) {
+        return (minimum + randomFloat(range));
+      } else {
+        return null;
+      }
+    }
+  }
+
+  public static class RandomDoubleGenerator extends RandomValueGenerator<Double> {
+    public RandomDoubleGenerator(long seed, int total, double minimum, double maximum) {
+      super(seed, PrimitiveType.PrimitiveTypeName.DOUBLE, total, minimum, maximum);
+    }
+
+    @Override
+    protected Double generateValue(Double minimum, Double maxiumum) {
+      double range = (maxiumum - minimum);
+      if (!shouldGenerateNull()) {
+        return (minimum + randomDouble(range));
+      } else {
+        return null;
+      }
+    }
+  }
+
+  public static class RandomBinaryGenerator extends RandomBinaryIterable<String> {
+    private static final int MAX_STRING_LENGTH = 16;
+    public RandomBinaryGenerator(long seed, int total) {
+      super(seed, PrimitiveType.PrimitiveTypeName.BINARY, total, MAX_STRING_LENGTH);
+    }
+
+    @Override
+    protected String generateValue(String minimum, String maxiumum) {
+      int stringLength = randomInt(15) + 1;
+
+      if (!shouldGenerateNull()) {
+        return randomString(stringLength);
+      } else {
+        return null;
+      }
+    }
+
+    @Override
+    public byte[] generateValueAsByteArray(String minimum, String maximum) {
+      String value = generateValue(minimum, maximum);
+      if (value != null) {
+        return value.getBytes();
+      } else {
+        return null;
+      }
+    }
+  }
+
+
+
+  public static class RandomFixedBinaryGenerator extends RandomBinaryIterable<String> {
+    public RandomFixedBinaryGenerator(long seed, int total, int length) {
+      super(seed, PrimitiveType.PrimitiveTypeName.FIXED_LEN_BYTE_ARRAY, total, length);
+    }
+
+    @Override
+    protected String generateValue(String minimum, String maxiumum) {
+      if (!shouldGenerateNull()) {
+        return randomFixedLengthString(bufferLength);
+      } else {
+        return null;
+      }
+    }
+
+    @Override
+    public byte[] generateValueAsByteArray(String minimum, String maximum) {
+      String value = generateValue(minimum, maximum);
+      if (value != null) {
+        return value.getBytes();
+      } else {
+        return null;
+      }
+    }
+  }
+
+
+
+  public static class RandomRange<T extends Comparable<T>> {
```

Comment:
All this does is validate that min is less than max. Do we need a class for this?